core/primitives-core/src/profile.rs

```diff
@@ -1,15 +1,18 @@
+use crate::config::{ActionCosts, ExtCosts};
+use borsh::{BorshDeserialize, BorshSerialize};
 use std::fmt;
 use std::ops::{Index, IndexMut};
+use strum::IntoEnumIterator;
 
-use borsh::{BorshDeserialize, BorshSerialize};
-
-use crate::config::{ActionCosts, ExtCosts};
-
+/// Serialization format to store profiles in the database.
+///
+/// This is not part of the protocol but archival nodes still rely on this not
+/// changing to answer old tx-status requests with a gas profile.
 #[derive(Clone, PartialEq, Eq)]
 pub struct DataArray(Box<[u64; Self::LEN]>);
 
 impl DataArray {
-    pub const LEN: usize = Cost::ALL.len();
+    pub const LEN: usize = if cfg!(feature = "protocol_feature_ed25519_verify") { 72 } else { 70 };
```

Comment:
Let's add a test that the `LEN` is correct, by comparing it with `iter().count()`

Reply:
It's not quite `iter().count()`, we have to dedup it first now that multiple costs map to the same entry. But you are right and I added a test to check this.